resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1193,6 +1189,37 @@ private[spark] class Client(
       }
   }
 
+  def init(): Unit = {
+    launcherBackend.connect()
+    // Setup the credentials before doing anything else,
+    // so we have don't have issues at any point.
+    setupCredentials()
+    yarnClient.init(yarnConf)
+    yarnClient.start()
+
+    setMaxNumExecutors()
+  }
+
+  /**
+   * If using dynamic allocation and user doesn't set spark.dynamicAllocation.maxExecutors
+   * then set the max number of executors depends on yarn cluster VCores Total.
+   * If not using dynamic allocation don't set it.
+   */
+  private def setMaxNumExecutors(): Unit = {
+    if (Utils.isDynamicAllocationEnabled(sparkConf)) {
+
+      val defaultMaxNumExecutors = DYN_ALLOCATION_MAX_EXECUTORS.defaultValue.get
+      if (defaultMaxNumExecutors == sparkConf.get(DYN_ALLOCATION_MAX_EXECUTORS)) {
+        val executorCores = sparkConf.getInt("spark.executor.cores", 1)
+        val maxNumExecutors = yarnClient.getNodeReports().asScala.
```

Comment:
Shouldn't we take queue's maxResources amount into account from [ResourceManager REST APIs](https://hadoop.apache.org/docs/r2.7.2/hadoop-yarn/hadoop-yarn-site/ResourceManagerRest.html)?